src/crownjewel/crownjewel.go

```diff
@@ -0,0 +1,392 @@
+
+		CrownjewelCronJob = nil
+	}
+}
+
+// Create Crown Jewel Policy based on K8s object type
+func CrownjewelPolicyMain() {
+	deployment := cluster.GetDeploymentsFromK8sClient()
+	client := cluster.ConnectK8sClient()
+
+	for _, d := range deployment {
+		err := getFilteredPolicy(client, d.Name, d.Namespace, d.Labels)
+		if err != nil {
+			log.Error().Msg("Error getting mount paths, err=" + err.Error())
+		}
+	}
+}
+
+type LabelMap = map[string]string
+
+// Get list of running processes from observability data
+func getProcessList(client kubernetes.Interface, namespace string, labels string) ([]string, error) {
+	var processList []string
+	duplicatePaths := make(map[string]bool)
+
+	podList, err := client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		LabelSelector: labels,
+	})
+	if err != nil {
+		log.Warn().Msg(err.Error())
+	}
+	for _, pod := range podList.Items {
+		for _, container := range pod.Spec.Containers {
+			sumResp, err := obs.GetSummaryData(&opb.Request{
+				PodName:       pod.Name,
+				NameSpace:     pod.Namespace,
+				ContainerName: container.Name,
+				Type:          "process,file",
+			})
+			if err != nil {
+				log.Warn().Msgf("Error getting summary data for pod %s, container %s, namespace %s: %s", pod.Name, container.Name, pod.Namespace, err.Error())
+				break
+			}
+
+			for _, fileData := range sumResp.FileData {
+				if !duplicatePaths[fileData.Source] {
+					// ignore serviceaccount related process
+					if strings.Contains(fileData.Destination, "serviceaccount") {
+						continue
+					}
+					processList = append(processList, fileData.Source)
+					duplicatePaths[fileData.Source] = true
+				}
+			}
+		}
+	}
+	return processList, nil
+}
+
+// Get all mounted paths
+func getVolumeMountPaths(client kubernetes.Interface, labels string) ([]string, error) {
+	podList, err := client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		LabelSelector: labels,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pod list: %v", err)
+	}
+
+	var mountPaths []string
+
+	for _, pod := range podList.Items {
+		for _, container := range pod.Spec.Containers {
+			for _, volumeMount := range container.VolumeMounts {
+				mountPaths = append(mountPaths, volumeMount.MountPath)
+			}
+		}
+	}
+	return mountPaths, nil
+}
+
+// Get used mount paths from observability data
+func usedMountPath(client kubernetes.Interface, namespace string, labels string) ([]string, map[string]string, error) {
+	var sumResponses []string
+	fromSource := make(map[string]string)
+
+	podList, err := client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		LabelSelector: labels,
+	})
+	if err != nil {
+		log.Warn().Msg(err.Error())
+	}
+
+	for _, pod := range podList.Items {
+		for _, container := range pod.Spec.Containers {
+			sumResp, err := obs.GetSummaryData(&opb.Request{
+				PodName:       pod.Name,
```

Comment:
> Why are we using pod name?

We are checking the resource to be pods (instead of deployment/statefulset etc.) because pod labels contains all the labels for that resource which might be missing in other workload (say statefulset for eg.).
Here below is an example of a vault statefulset:

Pod Labels:
```
Labels:     app.kubernetes.io/instance=vault
                  app.kubernetes.io/name=vault
                  component=server
                  controller-revision-hash=vault-5f4c59685d
                  helm.sh/chart=vault-0.24.1
                  statefulset.kubernetes.io/pod-name=vault-0
```
Statefulset labels:
```
  Labels:           app.kubernetes.io/instance=vault
                    app.kubernetes.io/name=vault
                    component=server
                    helm.sh/chart=vault-0.24.1```

Reply:
>  And how mount paths from observability data is utilized?

PTAL at https://github.com/accuknox/discovery-engine/pull/739#issuecomment-1741382140, if anything is unclear please comment.